libcudacxx/include/cuda/std/__cuda/semaphore.h

```diff
@@ -23,6 +24,19 @@
 
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA
 
+void simple_backoff()
```

Comment:
Critical: You notice that all our code is `__ugly` that is necessary to avoid clashes with user defined macros or other symbols. You need to change every variable function definition or whatnot to have a reserved identifier as the name.

Reply:
This also needs to run on device
```suggestion
_LIBCUDACXX_INLINE_VISIBILITY void simple_backoff() noexcept
```